ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -149,31 +156,39 @@ private[platform] class MutableCacheBackedContractStore(
         Future.successful(Some(contract))
       case Archived(stakeholders) if nonEmptyIntersection(stakeholders, readers) =>
         Future.successful(Option.empty)
-      case ContractStateValue.NotFound =>
-        logger.warn(s"Contract not found for $contractId")
-        Future.successful(Option.empty)
-      case existingContractValue: ExistingContractValue =>
+      case contractStateValue =>
+        // This flow is exercised when the readers are not stakeholders of the contract
+        // (the contract might have been divulged to the readers)
+        // OR the contract was not found in the index
+        //
+        // NOTE: The current implementation of mutable contract state cache provides an optimization aimed at performance-critical
+        //       applications that do not exercise this flow frequently (i.e. no negative contract lookups in command interpretation
+        //       or use of divulgence)
+        // TODO: Implement caching of divulged contracts for alleviating performance degradation in applications that
+        //       make use of divulgence OR applications that cause high numbers of negative contract lookups
```

Comment:
```suggestion
```
Move the note on the place where we don't cache negative lookups, as per my comment above.

I would not add the TODO, as it signals that we want to change that in the future, which we probably don't.